### PR TITLE
chore: tweak auto merging

### PR DESCRIPTION
**Pull request by mcdurdin**
Updates action versions, sets some delays for merge (because it was happening too quickly), cleans up yaml

### CODE REVIEW

File: .github/workflows/auto-approve-keyman-server-pr.yml [hunk 1]
```diff
@@ -7,15 +7,14 @@
...
+    permissions:
+      pull-requests: write
+    if: github.actor == 'keyman-server'
     steps:
-    - name: auto approve PR from keyman-server
-      uses: hmarr/auto-approve-action@v2.0.0
-      if: github.actor == 'keyman-server'
-      with:
-        github-token: "${{ secrets.GITHUB_TOKEN }}"
+      uses: hmarr/auto-approve-action@44888193675f29a83e04faf4002fa8c0b537b1e4 # v3.2.1
```
Comment (darcywong00) [inline .github/workflows/auto-approve-keyman-server-pr.yml:20]: github-token no longer needed?

Comment (darcywong00) [reply .github/workflows/auto-approve-keyman-server-pr.yml:20]: Is it cause the help site doesn't have [CODEOWNERS](https://github.com/hmarr/auto-approve-action#code-owners)?

Comment (mcdurdin) [reply .github/workflows/auto-approve-keyman-server-pr.yml:19]: apparently it's inherited from the environment by default.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**